test/Core.ts

```diff
@@ -384,45 +386,110 @@ export class Foo {
         })
       })
 
-      it('should replace imports from the project with the local source folder (double quotes)', async () => {
-        const state = prefixWithCwd({
-          'package.json': defaultPackageJson,
-          'src/utils/foo.ts': `
+      describe('Package Imports', () => {
+        type PackageName = { scope: Option<string>; name: string }
+
+        type PackageImport = {
+          quotes: 'single' | 'double'
+          packageName: PackageName
+          lib: boolean
+          path: Option<string>
+        }
+
+        const stringifyPackageName = ({ scope, name }: PackageName): string =>
+          `${O.fold(
+            () => '',
+            (str) => `@${str}/`
+          )(scope)}${name}`
+
+        const stringifyPackageImport = ({ quotes, packageName, lib, path }: PackageImport) => {
+          const quote = quotes === 'single' ? "'" : '"'
+          const libStr = lib ? '/lib' : ''
+          const pathStr = O.fold(
+            () => '',
+            (str) => `/${str}`
+          )(path)
+          const packageNameStr = stringifyPackageName(packageName)
+          return `${quote}${packageNameStr}${libStr}${pathStr}${quote}`
+        }
+
+        const mkState = (packageName: PackageName, importLine: string) =>
+          prefixWithCwd({
+            'package.json': JSON.stringify({ name: stringifyPackageName(packageName), homepage: 'http://foo' }),
+            'src/utils/foo.ts': `
 /**
  * @since 0.0.1
  */
-
 /**
  * @category utils
  * @since 0.0.1
  */
 export class Foo {
   /**
    * @example
-   * import * as Foo from "${JSON.parse(defaultPackageJson)['name']}"
+   * ${importLine}
```

Comment:
As we're testing also against scoped package names, different quotes, I think it's reasonable to parameterize the state creation.

Reply:
Agreed - seems reasonable to me.